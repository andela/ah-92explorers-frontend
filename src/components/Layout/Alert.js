/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/require-default-props */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Alert = ({ alerts }) => (
  alerts !== undefined
    && alerts !== null
    && alerts.length > 0
    && alerts.map(alert => (
      <div key={alert.id} className={`alert alert-${alert.alertType}`}>
        {alert.msg}
      </div>
    ))
);

Alert.propTypes = {
  alerts: PropTypes.array,
};

const mapStateToProps = state => ({
  alerts: state.alert,
});

export default connect(mapStateToProps)(Alert);
