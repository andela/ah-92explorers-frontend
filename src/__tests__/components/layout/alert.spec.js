import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import promiseMiddleware from 'redux-promise-middleware';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Alert from '../../../components/Layout/Alert';

const middlewares = [thunk, promiseMiddleware];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});
describe('<Alert />', () => {
  const props = {
    alerts: [{Math: 'ls'}]
  };
  it('should render Navbar component', () => {
    const wrapper = shallow(
        <Provider store={store}>
            <Alert {...props} />,
        </Provider>
      );
    expect(wrapper).toMatchSnapshot();
  });

});