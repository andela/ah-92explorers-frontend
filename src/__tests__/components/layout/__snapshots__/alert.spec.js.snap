// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render Navbar component 1`] = `ShallowWrapper {}`;
